src/Extensions/Activities/Internal/ActivityListenerInitializerService.cs

```diff
@@ -0,0 +1,77 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Omex.Extensions.Abstractions.Activities.Processing;
+
+namespace Microsoft.Omex.Extensions.Activities
+{
+	internal class ActivityListenerInitializerService : IHostedService
+	{
+		private readonly IActivityStartObserver[] m_activityStartObservers;
+		private readonly IActivityStopObserver[] m_activityStopObservers;
+		private readonly IActivityListenerConfigurator m_configurator;
+		private ActivityListener? m_listener;
+
+		public ActivityListenerInitializerService(
+			IEnumerable<IActivityStartObserver> activityStartObservers,
+			IEnumerable<IActivityStopObserver> activityStopObservers,
+			IActivityListenerConfigurator configurator)
+		{
+			m_activityStartObservers = activityStartObservers.ToArray();
+			m_activityStopObservers = activityStopObservers.ToArray();
+			m_configurator = configurator;
+		}
+
+		public Task StartAsync(CancellationToken cancellationToken)
```

Comment:
>cancellationToken [](start = 43, length = 17)

Does it make sense to check the cancellation token in the beginning?

Reply:
If I understand correctly in the case of canceled token canceled method won't be called, and since all operations quite fast here I think checking it won't help here it's mostly provided for cases if you are calling some async methods here.